deployment/databricks-connect.qmd

```diff
@@ -644,62 +637,94 @@ sc <- spark_connect(
+    that document, and then use its containing folder as the content's
+    location. Our hope is that for most time, this will be how you
+    deploy.
 
-``` r
-rsconnect::deployApp(
-  python = "Enter the Python environment to recreate",
-  envVars = c("DATABRICKS_HOST", "DATABRICKS_TOKEN"),
-  lint = FALSE
-)
-```
+-   Use the `appDir` argument to pass the folder path to be used.
+    Something such as `here::here("my-cool-document")` would work.
+
+-   If no document is opened in RStudio, and `appDir` is left empty,
+    then whatever `getwd()` returns is what will be used
+
+The aim of of the new function is to be as flexible, and helpful. It
+will gather your document location, credentials, and URLs. The
+**Python** location is what you will need to define at the time of
+calling `deploy_databricks()`. If you are in RStudio, and your document
+is opened, here are several ways to use:
+
+-   If you know the cluster's DBR version:
+
+    ``` r
+    pysparklyr::deploy_databricks(version = "14.1")
+    ```
 
-In the `python` argument, paste the value from the 'Path:' output
-`spark_connect()` used when you were testing your document locally. The
-`envVars` argument will let Posit Connect know to create those
-environment variables and to use the values from your machine.
+-   The cluster's ID can also be used, so `pysparklyr` pulls the DBR
+    version:
+
+    ``` r
+    pysparklyr::deploy_databricks(cluster_id = "1026-175310-7cpsh3g8")
+    ```
+
+-   If you just ran the code of the content you plan to deploy, then the
+    **Python** environment should already be loaded in the R session.
+    The function will validate that the path of that Python environment
+    conforms to one that `pysparklyr` creates, and uses that. This will
+    happen if no `version`, `cluster_id`, or `python` is passed. At that
+    point simply run:
+
+    ``` r
+    pysparklyr::deploy_databricks()
+    ```
+
+-   You can also pass the path to the Python environment to use by
+    setting the `python` argument:
 
-**Important:** Please note that `deployApp()` is not called from inside
-your document. It needs to be something you call outside, in your R
-console for example. If you wish to create an **R script** in your
-project that contains your deployment call, **do not place it in the
-same folder as your document**.
+    ``` r
+    pysparklyr::deploy_databricks(python = "/Users/edgar/.virtualenvs/r-sparklyr-databricks-14.1")
+    ```
 
-Here is a sample of a `deployApp()` call. There are some additional
-notes, as code comments, to further explain how and why to use:
+Here is an example of the output returned when using
+`deploy_databricks()`. Before submitting your content, you will be
+prompted to confirm that the information that it was able to gathered is
+correct. Also notice, that if you have more than one Posit Connect
+server setup in your RStudio IDE, it will choose the top one as the
+default, and let you easily change it by selecting, in this case 3:
```

Comment:
```suggestion
Here is an example of the output returned when using
`deploy_databricks()`. Before submitting your content, you will be
prompted to confirm that the information gathered is
correct. Also notice that if you have more than one Posit Connect
server setup in your RStudio IDE, it will choose the top one as the
default but allow you to easily change it if necessary:
```